src/disco/containers/InstallButton.js

```diff
@@ -29,12 +32,54 @@ export class InstallButton extends React.Component {
     downloadProgress: 0,
   }
 
+  componentDidMount() {
+    const { dispatch, guid, installURL, slug } = this.props;
+    this.addonManager = new AddonManager(guid, installURL, this.statusChanged);
+    this.addonManager.getAddon()
+      .then(() => {
+        dispatch({
+          type: 'INSTALL_STATE',
+          payload: {slug, guid, url: installURL, status: INSTALLED}});
+      })
+      .catch(() => {
```

Comment:
I'm thinking we should consider changing the `getAddon` code to return addon irrespective of whether the addon is present or not in other word just return `getAddonById()` That way all the dispatches can be done in an if statement in the `then` leaving the `catch` for actual errors. Let me know what you think.